packages/server/src/semester/semester.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@nestjs/common';
+import { Connection } from 'typeorm';
+import { SemesterModel } from './semester.entity';
+import { CourseModel } from '../course/course.entity';
+
+@Injectable()
+export class SemesterService {
+  constructor(private connection: Connection) {}
+
+  async setSemester(semester: SemesterModel, enable: boolean): Promise<void> {
```

Comment:
i think the name of this can be a little more descriptive. Maybe like `toggleActiveSemester` or something more apt for what it can be used for.